applications/RomApplication/tests/cpp_tests/test_rom_builder_and_solver.cpp

```diff
@@ -0,0 +1,220 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Eduard Gómez
+//
+
+// System includes
+
+/* External includes */
+
+/* Project includes */
+#include "containers/model.h"
+#include "geometries/triangle_2d_3.h"
+#include "includes/kratos_parameters.h"
+#include "testing/testing.h"
+#include "spaces/ublas_space.h"
+#include "solving_strategies/strategies/implicit_solving_strategy.h"
+#include "linear_solvers/linear_solver.h"
+#include "custom_strategies/rom_builder_and_solver.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos {
+namespace Testing {
+namespace ROMBuilderAndSolverTestingInternal {
```

Comment:
I'd have done this an anonymous namespace.

Reply:
I found that anonymous namespaces are dangerous when using unity builds since they can collide.

It was a while ago so I'm not 100% sure, but this was the rationale.

Reply:
OK. Argument makes sense for me.